keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,649 @@
+resources.
+
+## Motivation
+
+There are three major motivations for the solution in this KEP.
+
+* Enable the existing applications to run without modification.
+
+* Enable application developers, operators to transition to DRA gradually at
+  their own pace.
+
+* Enable cluster administrators to transition to DRA gradually at their own pace.
+
+For example, the following `Deployment` can be installed without modification on a
+cluster with DRA `ResourceSlice` below.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: demo
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: demo
+  template:
+    metadata:
+      labels:
+        app: demo
+    spec:
+      containers:
+      - name: demo
+        image: nvidia/cuda:8.0-runtime
+        command: ["/bin/sh", "-c"]
+        args: ["nvidia-smi && tail -f /dev/null"]
+        resources:
+          limits:
+            nvidia.com/gpu: 1
+```
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceSlice
+metadata:
+  name: gke-drabeta-n1-standard-4-2xt4-346fe653-zrw2-gpu.nvidia.coqj92d
+spec:
+  devices:
+  - basic:
+    name: gpu-0
+  - basic:
+    name: gpu-1
+  - basic:
+    name: gpu-2
+  - basic:
+    name: gpu-3
+  - basic:
+    name: gpu-4
+  - basic:
+    name: gpu-5
+  - basic:
+    name: gpu-6
+  - basic:
+    name: gpu-7
+  driver: gpu.nvidia.com
+  nodeName: gke-drabeta-n1-standard-4-2xt4-346fe653-zrw2
+```
+
+With this motivating example in mind. We define the following goals and
+non-goals of this KEP.
+
+### Goals
+
+* Introduce the ability for DRA to advertise extended resources listed in a
+  ResourceSlice, and for the scheduler to consider them for allocation.
+
+* Enable application operators to use the existing extended resource limits to
+request for DRA resources.
+
+### Non-Goals
+
+* Extended resource support is not added just for easing the transition to DRA
+  for the short term. Its ease of use is one big advantage to keep it remaining
+  useful for the long term.
+
+## Proposal
+
+The basic idea is the following:
+
+1. Introduce a field `ExtendedResourceName` to `ResourceSlice` and `Device` to
+   allow cluster administrators to configure DRA device driver to advertise
+   certain devices as extended resource.
+1. Introduce a special `ResourceClaim` to keep track of device allocations for
+   extended resource.
+
+With these additions in place, the scheduler has everything it needs to support
+the dynamic allocation of devices to requests made through extended resource.
+kubelet and DRA driver has everything it needs to admit and pass the allocated
+devices to the pod to run.
+
+## Design Details
+
+### Resource Slice API
+The exact set of proposed API changes on Resource Slice can be seen below:
+```go
+// ResourceSliceSpec contains the information published by the driver in one ResourceSlice.
+type ResourceSliceSpec struct {
+	...
+
+	// The extended resource name for all the devices in the ResourceSlice
```

Comment:
It would be helpful to include in the comments that the field on the device will override the name on the resuorceslice.

Reply:
done